src/query/storages/fuse/src/io/locations.rs

```diff
@@ -39,32 +39,52 @@ static SNAPSHOT_STATISTICS_V0: TableSnapshotStatisticsVersion =
 #[derive(Clone)]
 pub struct TableMetaLocationGenerator {
     prefix: String,
+    part_prefix: String,
 }
 
 impl TableMetaLocationGenerator {
     pub fn with_prefix(prefix: String) -> Self {
-        Self { prefix }
+        Self {
+            prefix,
+            part_prefix: "".to_string(),
+        }
+    }
+
+    pub fn with_part_prefix(mut self, part_prefix: String) -> Self {
+        self.part_prefix = part_prefix;
+        self
     }
 
     pub fn prefix(&self) -> &str {
         &self.prefix
     }
 
+    pub fn part_prefix(&self) -> &str {
+        &self.part_prefix
+    }
+
     pub fn gen_block_location(&self) -> (Location, Uuid) {
         let part_uuid = Uuid::new_v4();
-        (
-            (
-                format!(
-                    "{}/{}/{}_v{}.parquet",
-                    &self.prefix,
-                    FUSE_TBL_BLOCK_PREFIX,
-                    part_uuid.as_simple(),
-                    DataBlock::VERSION,
-                ),
+        let location_path = if self.part_prefix.is_empty() {
+            format!(
+                "{}/{}/{}_v{}.parquet",
+                &self.prefix,
+                FUSE_TBL_BLOCK_PREFIX,
+                part_uuid.as_simple(),
                 DataBlock::VERSION,
-            ),
-            part_uuid,
-        )
+            )
+        } else {
+            format!(
+                "{}/{}/{}_{}_v{}.parquet",
+                &self.prefix,
+                FUSE_TBL_BLOCK_PREFIX,
+                &self.part_prefix,
+                part_uuid.as_simple(),
+                DataBlock::VERSION,
+            )
+        };
+
+        ((location_path, DataBlock::VERSION), part_uuid)
```

Comment:
Could the prefix simply be `my_`?
So that we do not have to concatenate the prefix `my` and part name with a `_`.

Reply:
After discuss with @drmingdrmer , we decide:
1. when user input is `my`, we add a `_` and store in table meta.
2. when user input is `my_`, we no need add `_`.
3. after this we no need concat with `_` here.
cc @BohuTANG any suggestions?

Reply:
> * when user input is `my`, we add a `_` and store in table meta.
> * when user input is `my_`, we no need add `_`.

We can't predict user's data layout. User could have data like `myb_v2.parquet`.

In fact, I think this feature doesn't make sense. User rarely separate different data files based on prefix `mya`, `myb`. Instead, they will use `mya/`, `myb/`. This is what `root` designed for.

Users just need to use:

- `s3://bucket/path/to/mya/`
- `s3://bucket/path/to/myb/`

We don't need to support the case like `myaxxx_v2` or `mybxxx_v2`.

Reply:
We cannot assume that users store files in separate buckets, and that their upstream and downstream processing data may be based on different files inside the same bucket

Reply:
> We cannot assume that users store files in separate buckets

No, they are in the same bucket with different root.

`s3://bucket/path/to/mya/` => bucket: `bucket`, root: `/path/to/mya/`.

Reply:
I understand that this FEATURE is rarely used, but as said above, we can't assume user behavior.

Reply:
> I understand that this FEATURE is rarely used, but as said above, we can't assume user behavior.

If so, I prefer to not do any modification on user's input `prefix`. Just add them in our path, no matter user input `a` or `a_`, `a__`, `a_/_/_`.

Reply:
So far, the fuse engine has no dependency on the block filename, we are saving it to the segment file.